aurelia-slickgrid/src/aurelia-slickgrid/aurelia-slickgrid.ts

```diff
@@ -622,11 +626,51 @@ export class AureliaSlickgridCustomElement {
     this.grid.autosizeColumns();
   }
 
+  //
+  // private functions
+  // ------------------
+
+  /** Dispatch of Custom Event, which by default will bubble & is cancelable */
   private dispatchCustomEvent(eventName: string, data?: any, isBubbling: boolean = true, isCancelable = true): boolean {
     const eventInit: CustomEventInit = { bubbles: isBubbling, cancelable: isCancelable };
     if (data) {
       eventInit.detail = data;
     }
     return this.elm.dispatchEvent(new CustomEvent(eventName, eventInit));
   }
+
+  /** Load the Editor Collection asynchronously and replace the "collection" property when Promise resolves */
+  private loadEditorCollectionAsync(column: Column): any[] {
+    const collectionAsync = column && column.editor && column.editor.collectionAsync;
+    if (collectionAsync) {
+      // wait for the "collectionAsync", once resolved we will save it into the "collection"
+      // the collectionAsync can be of 3 types HttpClient, HttpFetch or a Promise
+      collectionAsync.then((response: HttpResponseMessage | Response | any[]) => {
+        if (response instanceof Response && typeof response['json'] === 'function') {
+          response['json']().then(data => this.updateEditorCollection(column, data));
```

Comment:
linters might warn or error here and recommed you use `response.json` over `response['json']. I know if you used the aurelia rules it would error

Reply:
inside the if 

```
if (response.bodyUsed) { 
  // do nothing or throw
} else response['json']().then(data => this.updateEditorCollection(column, data))

Reply:
what is the `bodyUsed` meant to do? As for the error, I think I would rather do nothing instead of throwing an error, can that ever happen anyway?

Reply:
`bodyUsed` checks to see if the response body was read from already. You can try it out in your code by calling `response.json` twice or i think if you call `response.text()` then `response.json()` too, both should error on the second call if i am not mistaken, (this would mimic what would happen if a user called `response.json` before your code),

Reply:
it can happen, but the user should protect against it if they know your code is reading it as well. Normally the user can prevent it by calling `response.clone()` first. I just did not know if you wanted to prevent a possible error from being thrown in your library.

Reply:
are you saying that we actually cannot call `.json()` twice? If so I didn't know and if that is the case then yes we should throw an error

Reply:
I would go with what you prefer since I'm not totally familiar with was is best to do here. If you could just provide me the lines of code then I'll update it tonight

Reply:
I can look more at it tomorrow and pull down ur code to double check what i have seen in the past. I would look at it tonight if I could, but I have not had much free time to code after work hours unfortunately

Reply:
just pushed another commit to change other things. BTW, since you will pull the code, I would say just push commit directly on the branch with the changes that make sense to you.

Reply:
just pushed the change with a comment for why in the commit message, sorry for the delay.